src/main/java/step05/domain/Car.java

```diff
@@ -0,0 +1,57 @@
+package step05.domain;
+
+import step03.strategy.MoveStrategy;
+
+import java.util.Objects;
+
+public class Car {
+    private final CarName name;
+    private final CarPosition position;
+
+    private Car(CarName carName, CarPosition carPosition) {
+        this.name = carName;
+        this.position = carPosition;
+    }
+
+    public static Car of(String name) {
+        return new Car(CarName.of(name), CarPosition.of());
+    }
+
+    public static Car of(String name, int position) {
+        return new Car(CarName.of(name), CarPosition.of(position));
+    }
+
+    public static Car of(String name, int position, int maxLength) {
+        return new Car(CarName.of(name, maxLength), CarPosition.of(position));
```

Comment:
```suggestion
        return this(CarName.of(name, maxLength), CarPosition.of(position));
```
요렇게 this 를 이용하여 다른 시그니쳐를 가진 생성자를 재사용하게 될 경우,
만약 생성자 레벨에서 추가적인 작업이 필요할 때 모든 생성자를 수정할 필요없이 root 생성자만 수정해주면 변경이 가능합니다 :)